services/services.go

```diff
@@ -809,6 +809,13 @@ func (sid *serviceID) Zone() string                      { return sid.zone }
 func (sid *serviceID) SetName(n string) ServiceID        { sid.name = n; return sid }
 func (sid *serviceID) SetEnvironment(e string) ServiceID { sid.env = e; return sid }
 func (sid *serviceID) SetZone(z string) ServiceID        { sid.zone = z; return sid }
+
+func (sid *serviceID) Equal(other ServiceID) bool {
```

Comment:
Not sure if you want to protect against nil here.

Reply:
sg, will do